### PR TITLE
force_ssl with tests

### PR DESCRIPTION
**Pull request by kalmanh**
This PR addresses https://github.com/simplabs/rails_api_auth/issues/29

A few comments:

1. The default is `force_ssl` like Marco requested
2. I am setting it to `false` so that the existing tests should pass
3. I am testing that when `force_ssl` is set to true or not set (default) - I get back a 301

### CODE REVIEW

File: app/controllers/oauth2_controller.rb [hunk 1]
```diff
@@ -5,6 +5,9 @@
 # @!visibility private
 class Oauth2Controller < ApplicationController
 
+  # Default to forcing SSL unless disabled
+  force_ssl if: -> { RailsApiAuth.force_ssl.nil? || RailsApiAuth.force_ssl }
```
Comment (marcoow) [inline app/controllers/oauth2_controller.rb]: I don't think the `nil` check is necessary here. Instead just use `!!RailsApiAuth.force_ssl` so that nil evaluates to `false`.

File: lib/rails_api_auth/engine.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
+require 'httparty'
```
Comment (marcoow) [inline lib/rails_api_auth/engine.rb]: This means that `httparty` will now always be required regardless of whether the `Facebook` or `Google` services are actually used or not. Also this change is unrelated to the SSL stuff.

File: spec/requests/oauth2_spec.rb [hunk 3]
```diff
@@ -1,11 +1,33 @@
...
+  before { RailsApiAuth.force_ssl = false }
+
   describe 'POST /token' do
     let(:params) { { grant_type: 'password', username: login.identification, password: login.password } }
 
     subject { post '/token', params }
 
+    context 'force_ssl' do
+      it 'responds with status 301 when not specified' do
+        RailsApiAuth.force_ssl = nil
```
Comment (marcoow) [inline spec/requests/oauth2_spec.rb]: see above - setting `force_ssl` to `nil` should disable SSL.

**Comment by kalmanh** [comment]
@marcoow - haha, I am confused.

1. Back in https://github.com/simplabs/rails_api_auth/issues/29#issuecomment-199315939 I suggested that the default for the `force_ssl` option would be `false`. You came back in the [next comment](https://github.com/simplabs/rails_api_auth/issues/29#issuecomment-199328555) and said that the default should be `true`. I understood that to mean that if nothing is specified in the initializer (default) SSL would be ON.  Now, you are saying "see above - setting force_ssl to nil should disable SSL". Can you clarify?

2. `RailsApiAuth.force_ssl.nil? || RailsApiAuth.force_ssl` means if it's nil or if it's TRUE. `!!RailsApiAuth.force_ssl` means if it's nil or if it's FALSE, so I don't think that would work as expected...

3. Yes, I will take the httparty stuff out (actually I will probably just do a new clean PR after closing this one) and we can "fight it out" :) in a new thread.

It could be that I am totally missing something here, but either way - please clarify

**Comment by marcoow** [comment]
> Back in #29 (comment) I suggested that the default for the force_ssl option would be false. You came back in the next comment and said that the default should be true. I understood that to mean that if nothing is specified in the initializer (default) SSL would be ON. Now, you are saying "see above - setting force_ssl to nil should disable SSL". Can you clarify?

The default should be `true` because we explicitly set it to `true` by default. Treating `nil` as truthy as well would be quite confusing for people.

> !!RailsApiAuth.force_ssl means if it's nil or if it's FALSE, so I don't think that would work as expected...

`!!RailsApiAuth.force_ssl` is `true` for truthy values and `false` otherwise - that's what we should use here

**Comment by kalmanh** [comment]
> The default should be true because we explicitly set it to true by default. 

OK, so maybe this is the part I am missing. In my mind - "default" means if nobody ever sets it. WHERE are you suggesting we set it to true? In the rails_api_auth engine or in the rails app using the engine? If in the app using the engine - that can hardly be called the "default".

**Comment by marcoow** [comment]
I'd say the default value is the value sth. has until it's changed ;) It should be set in `lib/rails_api_auth.rb`.

**Comment by kalmanh** [comment]
Makes sense. Closing this PR and will submit another one...

**Comment by kalmanh** [comment]
Rubocop doesn't like !!

http://www.rubydoc.info/github/bbatsov/rubocop/Rubocop/Cop/Style/DoubleNegation

**Comment by marcoow** [comment]
I'd be ok with disabling that check as it's a commonly used pattern

**Comment by kalmanh** [comment]
That's fine. `RailsApiAuth.force_ssl` works just as well as `!!RailsApiAuth.force_ssl`
